MediaBrowser.Providers/MediaInfo/AudioResolver.cs

```diff
@@ -0,0 +1,275 @@
+            }
+
+            var streams = GetExternalAudioStreams(video, 0, directoryService, clearCache);
+
+            foreach (var stream in streams)
+            {
+                yield return stream.Path;
+            }
+        }
+
+        public void AddExternalAudioStreams(
+            List<MediaStream> streams,
+            string videoPath,
+            int startIndex,
+            IReadOnlyList<string> files)
+        {
+            var videoFileNameWithoutExtension = NormalizeFilenameForAudioComparison(videoPath);
+
+            for (var i = 0; i < files.Count; i++)
+            {
+
+                var fullName = files[i];
+                var extension = Path.GetExtension(fullName.AsSpan());
+                if (!IsAudioExtension(extension))
+                {
+                    continue;
+                }
+
+                Model.MediaInfo.MediaInfo mediaInfo = GetMediaInfo(fullName).Result;
+                MediaStream mediaStream = mediaInfo.MediaStreams.First();
+                mediaStream.Index = startIndex++;
+                mediaStream.Type = MediaStreamType.Audio;
+                mediaStream.IsExternal = true;
+                mediaStream.Path = fullName;
+                mediaStream.IsDefault = false;
+                mediaStream.Title = null;
+
+                var fileNameWithoutExtension = NormalizeFilenameForAudioComparison(fullName);
+
+                // The audio filename must either be equal to the video filename or start with the video filename followed by a dot
+                if (videoFileNameWithoutExtension.Equals(fileNameWithoutExtension, StringComparison.OrdinalIgnoreCase))
+                {
+                    mediaStream.Path = fullName;
+                }
+                else if (fileNameWithoutExtension.Length > videoFileNameWithoutExtension.Length
+                         && fileNameWithoutExtension[videoFileNameWithoutExtension.Length] == '.'
+                         && fileNameWithoutExtension.StartsWith(videoFileNameWithoutExtension, StringComparison.OrdinalIgnoreCase))
+                {
+
+                    // Support xbmc naming conventions - 300.spanish.m4a
+                    var languageSpan = fileNameWithoutExtension;
+                    while (languageSpan.Length > 0)
+                    {
+                        var lastDot = languageSpan.LastIndexOf('.');
+                        var currentSlice = languageSpan[lastDot..];
+                        languageSpan = languageSpan[(lastDot + 1)..];
+                        break;
+                    }
+
+                    // Try to translate to three character code
+                    // Be flexible and check against both the full and three character versions
+                    var language = languageSpan.ToString();
+                    var culture = _localization.FindLanguageInfo(language);
+
+                    language = culture == null ? language : culture.ThreeLetterISOLanguageName;
```

Comment:
The language might be readily available in the mediainfo, so I would say that the filename should be a last resort